docs/datamaps/index.md

```diff
@@ -0,0 +1,192 @@
+    if (value != null) {
+        // And here we simply use the values
+        if (event.getPlayer().level().getRandom().nextFloat() > value.chance()) {
+            event.getPlayer().heal(value.amount());
+        }
+    }
+}
+```
+
+## Advanced data maps
+Advanced data maps are data maps which have added functionality. Namely, the ability of merging values and selectively removing them, through a remover. Implementing some form of merging and removers is highly recommended for data maps whose values are collection-likes (like `Map`s or `List`s).
+
+`AdvancedDataMapType` have one more generic besides `T` and `R`: `VR extends DataMapValueRemover<R, T>`. This additional generic allows you to datagen remove objects with increased type safety.
+
+### Creation
+You create an `AdvancedDataMapType` using `AdvancedDataMapType#builder`. Unlike the normal builder, the builder returned by that method will have two more methods (`merger` and `remover`), and it will return an `AdvancedDataMapType`. Registration methods remain the same.
+
+### Mergers
+An advanced data map can provide a `DataMapValueMerger` through `AdvancedDataMapType#merger`. This merger will be used to handle conflicts between data packs that attempt to attach a value to the same object.  
+The merger will be given the two conflicting values, and their sources (as an `Either<TagKey<R>, ResourceKey<R>>` since values can be attached to all entries within a tag, not just individual entries), and is expected to return the value that will actually be attached.  
+Generally, mergers should simply merge the values, and should not perform "hard" overwrites unless necessary (i.e. if merging isn't possible). If a pack wants to bypass the merger, it can do so by specifying the object-level `replace` field.  
+
+Let's imagine a scenario where we have a data map that attaches integers to items:
+```java
+public class IntMerger implements DataMapValueMerger<Item, Integer> {
+    @Override
+    public Integer merge(Registry<Item> registry, Either<TagKey<Item>, ResourceKey<Item>> first, Integer firstValue, Either<TagKey<Item>, ResourceKey<Item>> second, Integer secondValue) {
+        //highlight-next-line
+        return firstValue + secondValue;
+    }
+}
+```
+The above merger will merge the values if two datapacks attach to the same object. So if the first pack attaches the value `12` to `minecraft:carrot`, and the second pack attaches the value `15` to `minecraft:carrot`, the final value will be `27`. However, if the second pack specifies the object-level `replace` field, the final value will be `15` as the merger won't be invoked.
+
+We provide some default mergers for merging lists, sets and maps in `DataMapValueMerger`.  
+
+The default merger (`DataMapValueMerger#defaultMerger`) has the typical behaviour you'd expect from normal data packs, where the newest value (which comes from the highest datapack) always wins.
```

Comment:
"wins" is a weird term here. I'd use something like "where the newest value ... overwrites all values in datapacks 'below' it"